### PR TITLE
Added data engineering guides

### PR DESCRIPTION
**Pull request by mohi9282**
Resolves issue [#4472](https://github.com/ArcGIS/geosaurus/issues/4472) by adding 5 data engineering guide notebooks.

### CODE REVIEW

**Comment by AtmaMani** [comment]
@BP-Ent could you please review this PR?

**Comment by mohi9282** [comment]
@BP-Ent thanks for taking the time to review the guides.

@AtmaMani I have incorporated  the changes recommended by Breck. 


Attaching Breck's feedback doc here: 

[breck_review.docx](https://github.com/Esri/arcgis-python-api/files/5645974/breck_review.docx)

File: guide/16-data-engineering-guides/part1_introduction_to_dataengineering.ipynb [hunk 1]
```diff
@@ -0,0 +1,1637 @@
+{
```
Comment (AtmaMani) [inline guide/16-data-engineering-guides/part1_introduction_to_dataengineering.ipynb:1]: <p>In the first paragraph, we need to say that SeDF is a capability that is added to a regular Pandas DataFrame <strong>by the ArcGIS API for Python.</strong></p><p> </p><p>In the last sentence of this paragraph, we can add that what is shown below is just a quickstart to SeDF and in subsequent pages of the guide, SeDF will be dealt with in a detailed manner</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/842/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Comment by AtmaMani** [comment]
@mohi9282 thanks for the comprehensive primer for data engineering. I have asked for a minor text addition in notebook 1. @BP-Ent thanks for the detailed reviews (offline).

**Comment by mohi9282** [comment]
Hi @AtmaMani - I have made the suggested changes.

**Review by AtmaMani — APPROVED** [review]

**Pull request merged**